src/umpire/strategy/FixedPool.cpp

```diff
@@ -105,7 +105,6 @@ FixedPool::allocInPool(Pool& p) noexcept
     }
   }
 
-  UMPIRE_ASSERT("FixedPool: Logic error in allocate");
```

Comment:
Add an UMPIRE_ERROR here.

Reply:
Or change to:
```
UMPIRE_ASSERT("FixedPool: Logic error in allocate" && 0);
```

Reply:
Ah, upon a deeper check `allocInPool` should be `noexcept`, because only `allocate` is called and it has an assert check at the bottom before returning. Is that sufficient, @mcfadden8?

Reply:
I'm not sure I understand how `noexcept` is related to an `assert`.  I find it useful to have assertions happen closest to where things first started going wrong.

Reply:
I added back an error there.